api/src/test/java/jakarta/data/metamodel/NumericExpressionTest.java

```diff
@@ -199,4 +225,21 @@ void shouldCastToBigDecimal() {
             soft.assertThat(cast.type()).isEqualTo(BigDecimal.class);
         });
     }
+
+    @Test
+    @DisplayName("toString output must include parentheses where needed")
+    void shouldIncludeParenthesesInToString() {
+        NumericLiteral<Invoice, Integer> hundred = NumericLiteral.of(100);
+        // TODO this does not compile when the above is included directly
+        // within the following without the cast. Do literals need an of method
+        // that explicitly includes the entity class?
```

Comment:
We will need to fix that. Like what we did with `localDate()`, etc.

Reply:
Fix looks like it involves three changes:

```java
    static <N extends Number & Comparable<N>> NumericLiteral<Object,N> of(N value) {
        return new NumericLiteralRecord<>(value);
    }
```

```java
    Expression<? super T, N> left();

    Expression<? super T, N> right();
```

```java
record NumericOperatorExpressionRecord<T, N extends Number & Comparable<N>>
        (Operator operator, NumericExpression<? super T, N> left,
         NumericExpression<? super T, N> right)
```

We will have to audit stuff to see if there are any similar things hiding.

Reply:
It looks like you already have a PR open to cover thie above TODO (good idea to split it out from this one) so I'll review that.